packages/nuxt/src/runtime/plugins/sentry.client.ts

```diff
@@ -1,11 +1,41 @@
 import { getClient } from '@sentry/core';
-import { vueIntegration } from '@sentry/vue';
+import { browserTracingIntegration, vueIntegration } from '@sentry/vue';
 import { defineNuxtPlugin } from 'nuxt/app';
 
+// --- Types are copied from @sentry/vue (so it does not need to be exported) ---
+// The following type is an intersection of the Route type from VueRouter v2, v3, and v4.
+// This is not great, but kinda necessary to make it work with all versions at the same time.
+type Route = {
+  /** Unparameterized URL */
+  path: string;
+  /**
+   * Query params (keys map to null when there is no value associated, e.g. "?foo" and to an array when there are
+   * multiple query params that have the same key, e.g. "?foo&foo=bar")
+   */
+  query: Record<string, string | null | (string | null)[]>;
+  /** Route name (VueRouter provides a way to give routes individual names) */
+  name?: string | symbol | null | undefined;
+  /** Evaluated parameters */
+  params: Record<string, string | string[]>;
+  /** All the matched route objects as defined in VueRouter constructor */
+  matched: { path: string }[];
+};
+
+interface VueRouter {
+  onError: (fn: (err: Error) => void) => void;
+  beforeEach: (fn: (to: Route, from: Route, next?: () => void) => void) => void;
+}
+
 export default defineNuxtPlugin(nuxtApp => {
-  nuxtApp.hook('app:created', vueApp => {
-    const sentryClient = getClient();
+  const sentryClient = getClient();
```

Comment:
l: We can early return if client is not defined to remove the `if (sentryClient)` checks below.

Reply:
I'd rather leave the checks as I'm also accessing the client in a Nuxt lifecycle hook and I cannot be sure that the early return would hinder the lifecycle hook to be called if there is no client. Does this make sense?

Reply:
I see what you mean. I think then we need a new call to `getClient` inside the hook so we ensure that we always have an up-to-date reference.